### PR TITLE
feat: lazy persistent storage

### PR DESCRIPTION
**Pull request by Falynsky**
- added 'lazyPersistentStorage' parameter to @Cached()
- updated README.MD

### CODE REVIEW

File: packages/cached/lib/src/templates/cached_method_template.dart [hunk 1]
```diff
@@ -127,6 +127,10 @@ abstract class CachedMethodTemplate {
   String generateCacheMap() {
     final staticModifier = _getStaticModifier();
 
+    if (function.lazyPersistentStorage ?? false) {
```
Comment (LukaszWojcikIteo) [inline packages/cached/lib/src/templates/cached_method_template.dart:130]:
```suggestion
    if (_shouldUseLazyPersistentStorage) {
```

File: packages/cached/lib/src/templates/deletes_cache_method_template.dart [hunk 2]
```diff
@@ -49,8 +53,11 @@ class DeletesCacheMethodTemplate {
   }
 
   String _generateClearMaps() {
+    final persistemMethodsToClear =
```
Comment (LukaszWojcikIteo) [inline packages/cached/lib/src/templates/deletes_cache_method_template.dart:56]:
```suggestion
    final persistedMethodsToClear =
```

**Review by AndriiChemer — APPROVED** [review]

**Review by LukaszWojcikIteo — APPROVED** [review]

**Review by AndriiChemer — APPROVED** [review]

File: packages/cached/README.md [hunk 3]
```diff
@@ -553,6 +555,44 @@ Cached library supports usage of any external storage (e.g. Shared Preferences,
...
+  @Cached(lazyPersistentStorage: true)
+  Future<double> getLazyDouble() async {
+    return await _source.nextDouble() ;
+  }
+```
+
+When the `lazyPersistentStorage` parameter is used, it prevents the automatic loading of data from external storage into the cache managed by the caching library. For methods annotated with this parameter, the library's generator does not create a map for storing data fetched from the storage, nor does it initialize such a map before the method's invocation. Consequently, setting this parameter ensures that data is always fetched directly from externalStorage upon method call. If the data is not already present in externalStorage, it is retrieved and then stored there, as demonstrated in the provided example.
+
+```
+ @override
```
Comment (kamilraczka) [inline packages/cached/README.md:570]: I'd suggest to remove the generated code from README.md as much as we can - the above paragraph is fine but the generated code tends to change, it only adds "another place to amend" if we change something in the library and it is very easy to forget to update

**Comment by kamilraczka** [comment]
Do not merge things directly to master - create a proper release candidate for the changes

**Pull request merged**